dist/vesting_schedule_generator/index.js

```diff
@@ -0,0 +1,185 @@
+;
+const generateSchedule = (packagePath, equityCompensationIssuanceSecurityId) => {
+    const vestingSchedule = [];
+    const ocfPackage = (0, read_ocf_package_1.readOcfPackage)(packagePath);
+    const vestingTerms = ocfPackage.vestingTerms;
+    const transactions = ocfPackage.transactions;
+    const equityCompensationIssuance = transactions.find((transaction) => transaction.security_id === equityCompensationIssuanceSecurityId);
+    if (!equityCompensationIssuance) {
+        throw new Error(`Equity compensation issuance with security id ${equityCompensationIssuanceSecurityId} not found`);
+    }
+    const vestingStartTransaction = transactions.find((transaction) => transaction.object_type === "TX_VESTING_START" && transaction.security_id === equityCompensationIssuance.security_id);
+    const equityCompensationVestingTerms = vestingTerms.find((vestingTerm) => vestingTerm.id === equityCompensationIssuance.vesting_terms_id);
+    let transactionDate = new Date(vestingStartTransaction.date);
+    let vestingConditionId = vestingStartTransaction.vesting_condition_id;
+    let equityCompensationQuantity = parseFloat(equityCompensationIssuance.quantity);
+    let unvested = equityCompensationQuantity;
+    let vested = 0;
+    let currentVestingCondition = equityCompensationVestingTerms.vesting_conditions.find((vestingCondition) => vestingCondition.id === vestingConditionId);
+    if (currentVestingCondition.trigger.type !== "VESTING_START_DATE") {
+        throw new Error(`For this generator, the first condition must have a VESTING_START_DATE trigger.`);
+    }
+    let amountVested = (equityCompensationQuantity * parseFloat(currentVestingCondition.portion.numerator)) / parseFloat(currentVestingCondition.portion.denominator);
+    vested += amountVested;
+    unvested -= amountVested;
+    vestingSchedule.push({
+        Date: transactionDate.toISOString().split("T")[0],
+        "Event Type": "Start",
+        "Event Quantity": amountVested,
+        "Remaining Unvested": unvested,
+        "Cumulative Vested": vested,
+        "Cumulative Exercised": 0,
+        "Available to Exercise": vested,
+    });
+    vestingConditionId = currentVestingCondition.next_condition_ids[0];
+    currentVestingCondition = equityCompensationVestingTerms.vesting_conditions.find((vestingCondition) => vestingCondition.id === vestingConditionId);
+    if (currentVestingCondition.trigger.type !== "VESTING_SCHEDULE_RELATIVE") {
+        throw new Error(`This generator can only calculate for VESTING_SCHEDULE_RELATIVE triggers.`);
+    }
+    for (let i = 0; i < currentVestingCondition.trigger.period.occurrences; i++) {
+        transactionDate = new Date(transactionDate.setMonth(transactionDate.getMonth() + currentVestingCondition.trigger.period.length));
+        let cumulativePercent;
+        let lastCumulativePercent;
+        let remainder;
+        switch (equityCompensationVestingTerms.allocation_type) {
```

Comment:
What do we think about breaking the elements of this switch statement into separate functions to facilitate testing those as standalones?